src/Module/Api/Mastodon/Media.php

```diff
@@ -42,38 +42,37 @@ protected function post(array $request = [])
 
 		$type = Post\Media::getType($request['file']['type']);
 
-		if (in_array($type, [Post\Media::IMAGE, Post\Media::UNKNOWN])) {
+		if (in_array($type, [Post\Media::IMAGE, Post\Media::UNKNOWN, Post\Media::APPLICATION])) {
 			$media = Photo::upload($uid, $request['file'], '', null, null, '', '', $request['description']);
-			if (empty($media)) {
-				$this->logAndJsonError(422, $this->errorFactory->UnprocessableEntity());
+			if (!empty($media)) {
+				Logger::info('Uploaded photo', ['media' => $media]);
+				$this->jsonExit(DI::mstdnAttachment()->createFromPhoto($media['id']));
+			} elseif ($type == Post\Media::IMAGE) {
+				$this->jsonExit(DI::mstdnAttachment()->createFromPhoto($media['id']));
```

Comment:
@annando in this condition `$media` is empty, but will try to get `$media['id']`. Do you remember what was the purpose of this `elseif` statement?

refs #14830

Reply:
Not out of my head. I will have to have a deeper look.

Reply:
It seems to me as if the major part of this PR is the mimetype check and not this part of the code which looks odd to me. Please have a look at the Mastodon API description to check which HTTP error should be returned if the upload fails.

Reply:
On PUT we return a 'Record not found' error. I would do the same here:

```suggestion
			} else {
				$this->logAndJsonError(404, $this->errorFactory->RecordNotFound());
```

Reply:
It would of course also be a possibility to check the reason and to return a 422?

Reply:
At the moment we cannot really distinguish between the reasons, because the reason is logged inside the Model and an empty array is returned. In the future we should change this behavior to throw Exceptions inside the models, and catch and process (log and create a response) them in the Modules.

But for now the reasons justify 422 and 500, so I would go with the worst (500). The outcome will be the same: One has to check the logs for the reason of the failure.

Reply:
I will fix this in #14830 as proposed above.

Reply:
I don't really like the 500, since this is mostly thrown upon general PHP failures - which also means that when we can't insert the media, it is most likely some data problem, so 422 is a better solution here. I really would like to reduce the amount of 500 errors since this is a problem when you do support. I would like to have 500 only be thrown upon real PHP errors.

Reply:
Status codes 4xx are client errors, 5xx are server errors. Possible errors while a `Friendica\Model\Photo::uplaod()` call are:

- User with $uid not found: 500
- Empty upload file: 422
- No source file name: 422
- Image is invalid: 422
- Scaling picture to new size failed: 500
- Photo could not be stored: 500
- Photo not found: 500

As I mentioned there are multiple reasons for an error. As long as we cannot distinguish between the real error it is best practice to response with the more severe status code. Once we are starting with the refactoring of the Modules we will solve this problem everywhere.

Reply:
The problem with 500 is the support. When people report an error 500 when uploading pictures, it will get more complicated to help the users. Having the 500 solely for PHP errors would make support much easier. That's the reason why I really would like to not throw any 500 errors via the system at all. Is much easier to tell users "have a look at your PHP log and your PHP setup" when they report a 500 when we know that this error will only be thrown by the underlying system itself.

Reply:
Unconventional argument, but ok.
I'll change the error to `$this->logAndJsonError(422, $this->errorFactory->UnprocessableEntity());`.